test/test_support/geminabox_test_case.rb

```diff
@@ -235,14 +240,19 @@ def start_app!
     end
 
     @app_server = fork do
-      begin
-        Geminabox.data = config.data
+      SimpleCov.command_name SecureRandom.uuid
+      Geminabox.data = config.data
+      unless log_to_stdout?
         STDERR.reopen(File::NULL)
         STDOUT.reopen(File::NULL)
-        Rack::Server.start(server_options)
-      ensure
-        exit
       end
+      # Force kill our process to prevent ugly Webrick messages.
+      Signal.trap(1) do
+        # Data is only collected if we access the result.
+        SimpleCov.result
+        Process.kill(9, Process.pid)
+      end
+      Rack::Server.start(server_options)
```

Comment:
Is this really required to do like this kind of signal handling in tests?

Reply:
Yes. Previously a -9 was sent to the process, which shuts it down hard. This prevents SimpleCov to write its results to disk, so the coverage report is wrong. I you see a better way, please let me know.

Reply:
Thanks. Could you add the actual difference in coverage measurement with this change to make others understand easily?

Reply:
I'd be happy to oblige, but I'm not quite sure what you want me to document. Add screenshots to the PR? Convert them to text and make them part of the commit message? A lot things are not covered by unit tests but rely on integration tests. This change is required to collect coverage information from the server process. Without it, you get zero information.

Reply:
Now I got it. And I was afraid about changes in coverage but it now works well with test coverage in Code Climate for recent days (at that time of my initial review a day passed after #475 got merged, and no data on it).